### PR TITLE
Add openscap repo data and install openscap

### PR DESCRIPTION
**Pull request by shlomizadok**
What I'd like to achieve: __Installing OpenSCAP on the slave servers.__
How: Adding openscap repo data to the slave puppet.

~~Please note related PR: https://github.com/theforeman/puppet-foreman/pull/326
Which will require a version bump for foreman-infra/modules/foreman submodule.~~

### CODE REVIEW

**Comment by domcleal** [comment]
I'm not sure that depending on one class from that module is a good idea, we don't intend it to be reused like that.  e.g. it contains a reference to Foreman::Plugin in the middle.  Probably better just to copy the repo definition here.

**Comment by shlomizadok** [comment]
I am so happy you say that :)
I wanted not to DRY myself, though it seemed like a perfect situation were DRYing is acceptable :smile:

File: puppet/modules/slave/manifests/init.pp [hunk 1]
```diff
@@ -162,6 +162,30 @@
     }
   }
 
+  # Needed by foreman_openscap gem dependency OpenSCAP
+  if $osfamily == 'RedHat' {
+    $repo = $::operatingsystem ? {
+        'Fedora' => 'fedora',
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: don't worry about Fedora, we don't deploy it

File: puppet/modules/slave/manifests/init.pp [hunk 2]
```diff
@@ -162,6 +162,30 @@
...
+        default  => 'epel',
+      }
+
+      if $::operatingsystemmajrelease == undef {
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: don't worry about this, we have 1.7+ and can rely on it

File: puppet/modules/slave/manifests/init.pp [hunk 3]
```diff
@@ -162,6 +162,30 @@
...
+        $versions_array = split($::operatingsystemrelease, '\.') # facter 1.6
+        $major = $versions_array[0]
+      } else {
+        $major = $::operatingsystemmajrelease # facter 1.7+
+      }
+
+      yumrepo { 'isimluk-openscap':
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: indentation's off

File: puppet/modules/slave/manifests/init.pp [hunk 4]
```diff
@@ -162,6 +162,30 @@
...
+        enabled  => 1,
+        gpgcheck => 0,
+        baseurl  => "http://copr-be.cloud.fedoraproject.org/results/isimluk/OpenSCAP/${repo}-${major}-\$basearch/",
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: could you also set `includepkgs => ['openscap']` so we don't pick up other stuff from this repo?

File: puppet/modules/slave/manifests/init.pp [hunk 5]
```diff
@@ -162,6 +162,30 @@
...
+      } ->
+    package { 'openscap':
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: do we need -devel for compiling against?

File: puppet/modules/slave/manifests/init.pp [hunk 6]
```diff
@@ -162,6 +162,30 @@
...
+      ensure => '1.2.3'
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: trailing comma

**Comment by domcleal** [comment]
Yeah, DRY is good, but this isn't part of a shipped project so it's less important, especially crossing the module boundary...

**Comment by shlomizadok** [comment]
@domcleal, thank you so much for your comments. I hope it looks better now

File: puppet/modules/slave/manifests/init.pp [hunk 7]
```diff
@@ -162,6 +162,19 @@
...
 
+  # Needed by foreman_openscap gem dependency OpenSCAP
+  if $osfamily == 'RedHat' {
+    yumrepo { 'isimluk-openscap':
+      enabled  => 1,
+      gpgcheck => 0,
+      baseurl  => "http://copr-be.cloud.fedoraproject.org/results/isimluk/OpenSCAP/epel-${::operatingsystemmajrelease}-\$basearch/",
+      includepkgs => ['openscap'],
+    } ->
+    package { 'openscap':
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: should this be openscap-devel if you're compiling against it (e.g. https://github.com/theforeman/foreman-packaging/blob/rpm/develop/rubygem-openscap/rubygem-openscap.spec#L35)?  I assume you're adding this in order to build the openscap gem?

Comment (ohadlevy) [inline puppet/modules/slave/manifests/init.pp]: AFAIK no, as it needs to use FFI C functions.

Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: Ah yes, thanks.

File: puppet/modules/slave/manifests/init.pp [hunk 8]
```diff
@@ -162,6 +162,19 @@
...
+      ensure => '1.2.3',
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: This needs to be the full VR, so `1.2.3-0.1.el6` etc or just use `latest` (preferred).

File: puppet/modules/slave/manifests/init.pp [hunk 9]
```diff
@@ -162,6 +162,19 @@
     }
   }
 
+  # Needed by foreman_openscap gem dependency OpenSCAP
+  if $osfamily == 'RedHat' {
+    yumrepo { 'isimluk-openscap':
+      enabled  => 1,
+      gpgcheck => 0,
+      baseurl  => "http://copr-be.cloud.fedoraproject.org/results/isimluk/OpenSCAP/epel-${::operatingsystemmajrelease}-\$basearch/",
```
Comment (domcleal) [inline puppet/modules/slave/manifests/init.pp]: indentation's off on lines 168-170

**Comment by domcleal** [comment]
Thanks @shlomizadok, should happen within the next 40 mins or so if successful.

By the way, please leave a comment when you update a PR to say what's fixed, else I'll only find out it changed by luck or persistence.

**Pull request merged**